src/components/plugin-button/plugin-button.tsx

```diff
@@ -16,22 +16,34 @@ interface PluginButtonProps {
 }
 
 export const PluginButton = ({isActive, label, id, icon, dataTestId, setRef}: PluginButtonProps) => {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    setRef?.(buttonRef.current ?? null);
```

Comment:
this code is called once. what is the guarantee that buttonRef.current has a value at this point ?